lib/nxt_pipeline/error_callback.rb

```diff
@@ -0,0 +1,18 @@
+module NxtPipeline
+  class ErrorCallback
+    def initialize(errors, block)
+      @errors = errors
+      @block = block
+    end
+
+    attr_accessor :errors, :block
+
+    def applies_to_error?(error)
+      (error.class.ancestors & errors).any?
+    end
+
+    def call(step, arg, error)
```

Comment:
In general I like to have `&block` as an explicit argument in the method definition.

```suggestion
    def call(step, arg, error, &block)
```

Reply:
Totally agree. Usually always do that. I hate when methods take a block but don`t have it in the signature.